ubiquity/pay.ubq.fi
# Feat: Keychain Wizard

**👥 User Setup(secret.html):**

> WALLET_PRIVATE_KEY = Wallet's private key
> GITHUB_PAT = GitHub Personal Access Token (Classic) with org scope
> ORG_NAME = Organization Name
> EC_PUBLIC_KEY = public key (bot admin would provide a public key generated via keygen)

> 🔒 ENCRYPT - Encrypt the key first
> 🌐 SET - Setup the key on the repo
>

**🛡 Admin Setup(keygen.html):**

> 🔑 GENERATE - Generate new key-pair (X25519_PUBLIC_KEY = EC_PUBLIC_KEY)
> 🔒 ENCRYPT - Encrypt plain text using the public key
> 🔓 DECRYPT - Decrypt the cipher text using the private key
>

**🧪 Test cases:**

1. create a new org (if necessary)
2. generate a _PAT-Classic_ with **org** scope
3. get `EC_PUBLIC_KEY` or generate using _🔗keygen.html_
4. optionally enter advanced config in JSON/YAML
5. Encrypt the key first using `🔒 ENCRYPT` button
6. now use `🌐 SET` button to set those values to repo
7. check the target repo to verify (_🔗org/ubiquibot-config_)

Resolves #36
